### ISSUE TITLE
File Source error on Docker Apple M1: file IO: failed to initialize inotify

### ISSUE THREAD

**Issue by bobbyiliev** [MaterializeInc/materialize#9034]
### What version of Materialize are you using?

Or from the SQL console:

```
materialize=> select mz_version();
// Output
     mz_version      
---------------------
 v0.9.11 (0549e9e21)
(1 row)
```

### How did you install Materialize?

**Docker on Apple M1**

* [X] Docker image
* [ ] Linux release tarball
* [ ] APT package
* [ ] macOS release tarball
* [ ] Homebrew tap
* [ ] Built from source
* [ ] Materialize Cloud

### What was the issue?

After creating a file source, on Apple M1, materialized returns the following error when querying a materialized view:

```
ERROR:  Source error: materialize.public.requests: file IO: file source: failed to initialize inotify
```

### Is the issue reproducible? If so, please provide reproduction instructions.

You can reproduce the issue on Apple M1, by running the following demo:

https://github.com/bobbyiliev/mz-http-logs

The issue occurs when you try to query the materialized view:

```sql
materialize=> select * from unique_visitors;
// Output
ERROR:  Source error: materialize.public.requests: file IO: file source: failed to initialize inotify
```

Verified that the demo works fine on Ubuntu.

### Please attach any applicable log files.

```
materialized v0.9.11 (0549e9e21) listening on 0.0.0.0:6875...
2021-11-11T08:05:08.918669Z  INFO coord::catalog: create source materialize.public.requests (u1)    
2021-11-11T08:05:08.934863Z  INFO coord::timestamp: Timestamping Source u1 with Real Time Consistency.    
2021-11-11T08:05:16.352271Z  INFO coord::catalog: create view materialize.public.unique_visitors (u2)    
2021-11-11T08:05:16.352561Z  INFO coord::catalog: create index materialize.public.unique_visitors_primary_idx (u3)    
2021-11-11T08:05:16.484935Z ERROR dataflow::source::file: Failed to read file for u1/29. Error: file source: failed to initialize inotify. 
```

**Comment by philip-stoev** [comment]
Apparently ionotify does not work under qemu under M1 . Other projects are running into the same issue https://www.google.com/search?q=docker+m1+ionotify&oq=docker+m1+ionotify&aqs=chrome..69i57j0i13.3083j0j7&client=ubuntu-browser&sourceid=chrome&ie=UTF-8

The stop-gap solution would be to fail-back to looping instead of ionotify when checking if the file has been appended to. The long-term solution would be to provide native Arm docker images, which is https://github.com/MaterializeInc/materialize/issues/8140

The stop-gap solution would make it a good starter issue for new hires that have received a M1 laptop. So I am tagging it as such.

**Comment by benesch** [comment]
Thanks for the report, @bobbyiliev! This has been fixed in v0.11.0 with our native ARM binaries.
